datadog_checks_base/tests/test_db_statements.py

```diff
@@ -109,6 +113,137 @@ def key(row):
         assert 1 == len(sm.compute_derivative_rows(rows3, metrics, key=key))  # only 1 row computed
         assert 2 == len(sm.compute_derivative_rows(rows4, metrics, key=key))  # both rows computed
 
+    def test_compute_derivative_rows_with_duplicates(self):
+        sm = StatementMetrics()
+
+        def key(row):
+            return (row['query_signature'], row['db'], row['user'])
+
+        metrics = ['count', 'time']
+
+        rows1 = [
+            {
+                'count': 13,
+                'time': 2005,
+                'errors': 1,
+                'query': 'SELECT * FROM table1 where id = ANY(?)',
+                'query_signature': 'sig1',
+                'db': 'puppies',
+                'user': 'dog',
+            },
+            {
+                'count': 25,
+                'time': 105,
+                'errors': 0,
+                'query': 'SELECT * FROM table1 where id = ANY(?, ?)',
+                'query_signature': 'sig1',
+                'db': 'puppies',
+                'user': 'dog',
+            },
+        ]
+
+        rows2 = [
+            {
+                'count': 14,
+                'time': 2006,
+                'errors': 2,
+                'query': 'SELECT * FROM table1 where id = ANY(?)',
+                'query_signature': 'sig1',
+                'db': 'puppies',
+                'user': 'dog',
+            },
+            {
+                'count': 26,
+                'time': 106,
+                'errors': 1,
+                'query': 'SELECT * FROM table1 where id = ANY(?, ?)',
+                'query_signature': 'sig1',
+                'db': 'puppies',
+                'user': 'dog',
+            },
+        ]
+
+        # Run a first check to initialize tracking
+        sm.compute_derivative_rows(rows1, metrics, key=key)
+        # Run the check again to compute the metrics
+        metrics = sm.compute_derivative_rows(rows2, metrics, key=key)
+
+        expected_merged_metrics = [
+            {
+                'count': 2,
+                'time': 2,
```

Comment:
nit: generally better to pick different numbers for expected test results. We could swap count and time values by accident and the test would not catch it.

Reply:
That was a great observation. I updated the values to be different and address this.